### PR TITLE
Add max tree height and card height to VirtualizedTree props.

### PR DESCRIPTION
**Pull request by leslielazzarino**
Signed-off-by: Leslie Lazzarino <leslie.lazzarino@tngtech.com>

### Summary of changes

maxTreeHeight and cardHeight are now properties of the VirtualizedTree, instead of being hard coded somewhere inside. Also the naming scheme in RenderTree has been made more consistent, avoiding the use of terms like "folder" and "file" where expandable/simple nodes are meant.
 
### Context and reason for change

Making VirtualizedTree more generic for package extraction.

### How can the changes be tested

Basically a refactoring, the file tree in the app should work like before.

Note: Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

### CODE REVIEW

**Review by nicarl — APPROVED** [review]

File: src/Frontend/Components/ResourceBrowser/ResourceBrowser.tsx [hunk 1]
```diff
@@ -97,6 +99,8 @@ export function ResourceBrowser(): ReactElement | null {
       );
   }
 
+  const treeMaxHeight: number = useWindowHeight() - topBarHeight - 4;
```
Comment (benedikt-richter) [inline src/Frontend/Components/ResourceBrowser/ResourceBrowser.tsx:102]: treeMaxHeight -> maxTreeHeight or simply maxHeight

File: src/Frontend/Components/ResourceBrowser/ResourceBrowser.tsx [hunk 2]
```diff
@@ -107,6 +111,8 @@ export function ResourceBrowser(): ReactElement | null {
       selectedResourceId={selectedResourceId}
       ariaLabel={'resource browser'}
       getTreeItemLabel={getTreeItemLabelGetter()}
+      cardHeight={20}
```
Comment (benedikt-richter) [inline src/Frontend/Components/ResourceBrowser/ResourceBrowser.tsx:114]: `20` should be extracted as constant

File: src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx [hunk 3]
```diff
@@ -93,8 +93,14 @@ export function VirtualizedTree(
...
           length={treeItems.length}
-          max={{ height: treeHeight }}
-          cardVerticalDistance={20}
+          max={
+            props.treeMaxHeight
+              ? { height: props.treeMaxHeight }
+              : {
+                  numberOfDisplayedItems: min([treeItems.length, 5]) as number,
+                }
+          }
```
Comment (benedikt-richter) [inline src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx:102]: could be extracted

File: src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx [hunk 4]
```diff
@@ -93,8 +93,14 @@ export function VirtualizedTree(
       <div className={classes.content}>
         <List
           length={treeItems.length}
-          max={{ height: treeHeight }}
-          cardVerticalDistance={20}
+          max={
+            props.treeMaxHeight
+              ? { height: props.treeMaxHeight }
+              : {
+                  numberOfDisplayedItems: min([treeItems.length, 5]) as number,
```
Comment (benedikt-richter) [inline src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx:100]: `5`should be extracted as constant

File: src/Frontend/Components/VirtualisedTree/render-tree.tsx [hunk 5]
```diff
@@ -137,7 +138,8 @@ function containsExactlyOneFolder(
   containedNodes: Array<string>
 ): boolean {
   return (
-    containedNodes.length == 1 && canHaveChildren(resource[containedNodes[0]])
+    containedNodes.length == 1 &&
```
Comment (benedikt-richter) [inline src/Frontend/Components/VirtualisedTree/render-tree.tsx:141]: `==` -> `===`

**Review by benedikt-richter — CHANGES_REQUESTED** [review]
see comments

**Review by benedikt-richter — APPROVED** [review]
